cloud/amazon/ec2_vpc_peer.py

```diff
@@ -0,0 +1,1544 @@
+                accept_with_profile, region, accepter_routes,
+                requester_routes, check_mode
+            )
+        )
+        if success:
+            err_msg = ''
+
+    results = convert_to_lower(results)
+    if results.get('tags', None):
+        results['tags'] = convert_list_of_tags(results['tags'])
+
+    return success, changed, err_msg, results
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            accepter_routes=dict(type='list'),
+            requester_routes=dict(type='list'),
+            vpc_id=dict(),
+            vpc_peer_id=dict(),
+            vpc_peering_id=dict(),
+            peer_owner_id=dict(),
+            accept_peer=dict(type='bool', default=False),
+            profile=dict(),
+            accept_with_profile=dict(),
+            resource_tags=dict(type='dict'),
+            state=dict(
+                default='present', choices=[
+                    'present', 'absent', 'accept', 'reject'
+                ]
+            )
+        )
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 is required.')
+
+    check_mode = module.check_mode
+    accept_with_profile = module.params.get('accept_with_profile')
+    accepter_routes = module.params.get('accepter_routes')
+    requester_routes = module.params.get('requester_routes')
+    boto_profile = module.params.get('profile')
+    vpc_id = module.params.get('vpc_id')
+    vpc_peer_id = module.params.get('vpc_peer_id')
+    vpc_peering_id = module.params.get('vpc_peering_id')
+    peer_owner_id = module.params.get('peer_owner_id')
+    accept_peer = module.params.get('accept_peer')
+    tags = module.params.get('resource_tags')
+    state = module.params.get('state').lower()
+
+    if tags:
+        tags = make_tags_in_aws_format(tags)
+
+    if state == 'present' and not tags:
+        err_msg = "parameters state=present and tags are required together"
+        module.fail_json(
+            success=False, changed=False, result={}, msg=err_msg
+        )
+
+    if accept_with_profile and state == 'present' and not accept_peer:
+        err_msg = "accept_with_profile can only be used with accept_peer"
+        module.fail_json(
+            success=False, changed=False, result={}, msg=err_msg
+        )
+
+    try:
+        region, ec2_url, aws_connect_kwargs = (
+            get_aws_connection_info(module, boto3=True)
+        )
+        client = (
+            boto3_conn(
+                module, conn_type='client', resource='ec2',
+                region=region, endpoint=ec2_url, **aws_connect_kwargs
+            )
+        )
+    except botocore.exceptions.ClientError, e:
+        err_msg = 'Boto3 Client Error - {0}'.format(str(e.msg))
+        module.fail_json(
+            success=False, changed=False, result={}, msg=err_msg
```

Comment:
Handle boto3 exceptions as per the guidelines https://github.com/ansible/ansible-modules-extras/blob/devel/cloud/amazon/GUIDELINES.md#boto3-2